arangod/Graph/Queues/FifoQueue.h

```diff
@@ -51,6 +51,7 @@ class FifoQueue {
   };
 
   void append(Step step) {
+    // TODO: We need to implement a method to verify that we do not have duplicate looseEnds objects here (e.g. same vertex looseEnd twice)
```

Comment:
```suggestion
    // TODO: We need to implement a method to verify that we do not have duplicate looseEnds objects here (e.g. same vertex looseEnd twice)
```

It is not the job of the FIFO queue to make sure the paths are unique.
Needs to be handled outside.